integration/test/ParseQueryTest.js

```diff
@@ -1661,4 +1663,72 @@ describe('Parse Query', () => {
       done();
     });
   });
+
+  describe('Parse Query Pinning (Default)', () => {
+    beforeEach((done) => {
+      const DefaultStorageController = require(path.resolve(__dirname, '../../lib/node/LocalDatastoreController.default'));
+      Parse.CoreManager.setLocalDatastoreController(DefaultStorageController);
+      done();
+    });
+
+    it('can query from pin with name', (done) => {
+      const obj1 = new TestObject({ field: 1 });
+      const obj2 = new TestObject({ field: 2 });
+      const obj3 = new TestObject({ field: 3 });
+      const item = new Item();
+      const objects = [obj1, obj2, obj3, item];
+      Parse.Object.saveAll(objects).then(() => {
+        Parse.Object.pinAllWithName('test_pin', objects);
+        const query = new Parse.Query(TestObject);
+        query.greaterThan('field', 1);
+        query.fromPinWithName('test_pin');
+        return query.find();
+      }).then((results) => {
+        assert.equal(results.length, 2);
+        assert(results[0].get('field') > 1);
+        assert(results[1].get('field') > 1);
+        done();
+      });
+    });
+
+    it('can query from local datastore', (done) => {
+      const obj1 = new TestObject({ field: 1 });
+      const obj2 = new TestObject({ field: 2 });
+      const obj3 = new TestObject({ field: 3 });
+      const objects = [obj1, obj2, obj3];
+      Parse.Object.saveAll(objects).then(() => {
+        Parse.Object.pinAll(objects);
+        const query = new Parse.Query(TestObject);
+        query.fromLocalDatastore();
+        return query.find();
+      }).then((results) => {
+        assert.equal(results.length, 3);
+        done();
+      });
+    });
+
+    it('can query from pin', (done) => {
+      const obj1 = new TestObject({ field: 1 });
+      const obj2 = new TestObject({ field: 2 });
+      const obj3 = new TestObject({ field: 3 });
+      const objects = [obj1, obj2, obj3];
+      Parse.Object.saveAll(objects).then(() => {
+        Parse.Object.pinAll(objects);
+        const query = new Parse.Query(TestObject);
+        query.fromPin();
+        return query.find();
+      }).then((results) => {
+        assert.equal(results.length, 3);
+        done();
+      });
+    });
+  });
+
+  describe('Parse Query Pinning (LocalStorage)', () => {
```

Comment:
@flovilmart Quick question? Since I have both in-memory and localStorage available. I want to run the test cases against both without copying them. Would putting describe inside of a for-loop make sense?

Reply:
Yep it works, otherwise what you can do is put it into a functions and call the function from the specific describe

Reply:
I just tried putting it in a loop. It also works great with Jasmine Test Explorer too